ale_python_interface/ale_c_wrapper.cpp

```diff
@@ -1,3 +1,20 @@
-//all code is currently in the .h file
 #include "ale_c_wrapper.h"
 
+#include <cstring>
+#include <string>
+
+void encodeState(ALEState *state, char *buf) {
+	std::string str = state->serialize();
+
+	memcpy(buf, str.data(), str.length());
```

Comment:
Ah, right. Can you also pass in the buffer size for safety, & throw an exception as needed?